src/Agent.Worker/Build/BuildDirectoryManager.cs

```diff
@@ -122,7 +122,14 @@ public sealed class BuildDirectoryManager : AgentService, IBuildDirectoryManager
             // Set the default clone path for each repository (the Checkout task may override this later)
             foreach (var repository in repositories)
             {
-                var repoPath = GetDefaultRepositoryPath(executionContext, repository, newConfig.SourcesDirectory);
+                var repoSourceDirectory = newConfig?.RepositoryTrackingInfo.Where(item => string.Equals(item.Identifier, repository.Alias, StringComparison.OrdinalIgnoreCase)).Select(item => item.SourcesDirectory).FirstOrDefault();
+
+                if (repoSourceDirectory == null)
+                {
+                    throw new ArgumentNullException($"Repository with alias {repository.Alias} wasn't found in new tracking config");
+                }
+
+                var repoPath = GetDefaultRepositoryPath(executionContext, repository, repoSourceDirectory);
```

Comment:
We don't want to change the existing behavior of single repo checkouts, which is to use the `s` directory without creating a subdirectory. None of the paths in the tests should have needed to be updated. I think the logic you need here (eliminating the `GetDefaultRepositoryPath` method, since it makes it awkward to express the logic to fix this bug) is:
```csharp
String repoPath;
if (RepositoryUtil.HasMultipleCheckouts(executionContext.JobSettings))
{
    // If we have multiple checkouts they should all be rooted to the sources directory (_work/1/s/repo1)
    var repoSourceDirectory = newConfig?.RepositoryTrackingInfo.Where(item => string.Equals(item.Identifier, repository.Alias, StringComparison.OrdinalIgnoreCase)).Select(item => item.SourcesDirectory).FirstOrDefault();
    if (repoSourceDirectory != null)
    {
        repoPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), repoSourceDirectory);
    }
    else
    {
        repoPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), newConfig.SourcesDirectory, RepositoryUtil.GetCloneDirectory(repository));
    }
}
else
{
    // For single checkouts, the repository is rooted to the sources folder (_work/1/s)
    repoPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), newConfig.SourcesDirectory);
}
```
That should maintain the existing location for single repos while loading the correct default from the tracking context for multi repos.

Reply:
@mjroghelia I made changes based on your review, could you please take a look at PR once again?

Reply:
Thanks